src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -440,15 +440,21 @@ private static string ComposeMessageFromInnerExceptions(Exception exception)
                     package.HasCompileTimePlaceholder &&
                     !TargetFramework.IsPackageBased)
                 {
+                    // requiresFrameworkAssemblies is true whenever we find a CompileTimePlaceholder in a non-package based framework, even if
+                    // the reference is unresolved. This ensures the best error experience when someone is building on a machine without
+                    // the target framework installed.
+                    requiresFrameworkAssemblies = true;
```

Comment:
Setting this to true enables the check to make sure the target framework is installed on the machine.

@davidfowl and I discussed maybe this rule changes to just be `!TargetFramework.IsPackageBased`, instead of also needing a `HasCompileTimePlaceholder` package.